query-engine/core/src/query_document/error.rs

```diff
@@ -1,28 +1,73 @@
 use crate::{schema::InputType, ArgumentValue};
 use fmt::Display;
 use itertools::Itertools;
+use serde::Serialize;
 use std::fmt;
+use user_facing_errors::UserFacingError;
 
 #[derive(Debug)]
-pub struct QueryParserError {
-    pub path: QueryPath,
-    pub error_kind: QueryParserErrorKind,
+pub enum QueryParserError {
+    Structured(StructuredQueryParseError),
+    Legacy {
```

Comment:
This will only last during the transitional process of the refactoring; at some point, there won't be any Legacy variant, and the `Structured` variant and `StructuredQueryParseError` will be inlined, such as `StructuredQueryParseError` becomes `QueryParserError` but now implementing